src/ucp/api/ucp.h

```diff
@@ -2177,8 +2177,112 @@ ucs_status_t ucp_worker_fence(ucp_worker_h worker);
+ */
+ucs_status_t ucp_atomic_post(ucp_ep_h ep, ucp_atomic_post_op_t opcode, uint64_t value,
+                             size_t op_size, uint64_t remote_addr, ucp_rkey_h rkey);
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Post an atomic fetch operation.
+ *
+ * This routine will post an atomic fetch operation to remote memory.
+ * The remote value is described by the combination of the remote
+ * memory address @a remote_addr and the @ref ucp_rkey_h "remote memory handle"
+ * @a rkey.
+ * The routine is non-blocking and therefore returns immediately. However the 
+ * actual atomic operation may be delayed. The atomic operation is not considered complete
+ * until the values in remote and local memory are completed. If the atomic operation 
+ * operation is completed immediately the routine returns UCS_OK and the 
+ * call-back routine @a cb is @b not invoked. If the operation is @b not 
+ * completed immediately and no error reported then the UCP library will 
+ * schedule to invoke the call-back routine @a cb whenever the atomic operation will be
+ * completed. In other words, the completion of an atomic operation can be signaled by 
+ * the return code or the call-back.
+ *
+ * @note The user should not modify any part of the @a result after this
+ *       operation is called, until the operation completes.
+ *
+ * @param [in] ep          UCP endpoint.
+ * @param [in] opcode      One of @ref ucp_atomic_fetch_op_t.
+ * @param [in] value       Source operand for atomic operation. In the case of CSWAP
+ *                         this is the conditional for the swap. For SWAP this is
+ *                         the value to be placed in remote memory.
+ * @param [inout] result   Local memory address to store resulting fetch to.
+ *                         In the case of CSWAP the value in result will be
+ *                         swapped into the @ref remote_addr if the condition
+ *                         is true.
+ * @param [in] op_size     Size of value in bytes and pointer type for result
+ * @param [in] remote_addr Remote address to operate on.
+ * @param [in] rkey        Remote key handle for the remote address.
+ * @param [in] cb          Call-back function that is invoked whenever the
+ *                         send operation is completed. It is important to note
+ *                         that the call-back function is only invoked in a case when
+ *                         the operation cannot be completed in place.
+ *
+ * @return UCS_OK               - The operation was completed immediately.
+ * @return UCS_PTR_IS_ERR(_ptr) - The operation failed.
+ * @return otherwise            - Operation was scheduled and can be
+ *                              completed at any point in time. The request handle
+ *                              is returned to the application in order to track
+ *                              progress of the operation. The application is
+ *                              responsible to released the handle using
```

Comment:
s/responsible to released/responsible for releasing/

This wording looks to have been cut-n-pasted from the ucp_tag_* descriptions.  So, at some point, this wording should be corrected there, as well (just a note for future cleanup).